manager/controlapi/secret.go

```diff
@@ -12,13 +17,40 @@ import (
 // MaxSecretSize is the maximum byte length of the `Secret.Spec.Data` field.
 const MaxSecretSize = 500 * 1024 // 500KB
 
+var isValidSecretName = regexp.MustCompile(`^[a-zA-Z0-9](?:[a-zA-Z0-9-_.]{0,62}[a-zA-Z0-9])*$`)
```

Comment:
`validSecretNameRegexp`.

Describe what this regexp is _supposed_ to do.

Looks like we can structure this in a more simple manner:

```
^[a-zA-Z0-9]+(?:[-_.]{1}[a-zA-Z0-9]+)*$
```

Validate total length outside; regexp isn't good at doing length checks. As they change, maintaining the length property may break down.

Also, consider not starting with numbers.